control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -169,7 +176,15 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager, create
 	r.SetDefaultSecurityContext = !r.ManagementClusterCapabilities.Has(capabilities.CapabilitySecurityContextConstraint)
 
 	r.reconcileInfrastructureStatus = r.defaultReconcileInfrastructureStatus
+	r.createAWSSecurityGroup = createAWSDefaultSecurityGroup
 
+	// AWS_SHARED_CREDENTIALS_FILE and AWS_REGION envvar should be set in operator deployment
+	// when reconciling an AWS hosted control plane
+	if os.Getenv("AWS_SHARED_CREDENTIALS_FILE") != "" {
```

Comment:
what if the sts token gets rotated? wouldn't then the client become useless by instantiating it only once here?

Reply:
My understanding is that the token is not read until it's actually used by a client call. We store the ec2Client for the endpoint service reconciler as well: https://github.com/openshift/hypershift/blob/73b5cc157375233069f96897db27b05ff34ac71c/control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go#L178-L182

Reply:
You're right, for the record: 
NewSession -> NewSessionWithOptions ->  newSession ->  mergeConfigSrcs ->  resolveCredentials -> assumeWebIdentity -> Stores then TokenFetcher function with the file path.

Then this is token fetcher is called by a function which is call every request via InjectHandlers.